### PR TITLE
Pure variant packages

### PR DESCRIPTION
**Pull request by tttppp**
Use packages to store information about each variant.

### CODE REVIEW

**Comment by tttppp** [comment]
This includes everything in the PureVariant pull request. I didn't want to move all the code in the same pull request in case it made reviewing harder.

File: classical/start/map.go [hunk 1]
```diff
@@ -183,3 +183,22 @@ func Graph() *graph.Graph {
 		Prov("kie").Conn("hol", c.Coast...).Conn("hel", c.Sea).Conn("den", c.Coast...).Conn("bal", c.Sea).Conn("ber", c.Coast...).Conn("mun", c.Land).Conn("ruh", c.Land).Flag(c.Coast...).SC(c.Germany).
 		Done()
 }
+
+func PureGraph() *graph.Graph {
```
Comment (zond) [inline classical/start/map.go]: Shouldn't this be in the pure variant directory?

Comment (tttppp) [inline classical/start/map.go]: Yes it should.

**Comment by tttppp** [comment]
I guess classical is just a variant that a lot of other variants depend on. I don't think we want variant directories at the top level, as in the future they might cause quite a bit of clutter.

**Comment by zond** [comment]
Fair enough

**Comment by zond** [comment]
So, what's your plan now? Tell me about the order you want your PRs merged?

**Comment by zond** [comment]
This doesn't run the tests, FYI - a lot of references to the previous location of the moved packages in the source files.

**Comment by zond** [comment]
Tell me when you're happy with this, and I'll test and merge it!

**Comment by tttppp** [comment]
Ok - I just ran quite a few of the tests and they all seem ok. This needs to be merged at the same time as the diplicity "PureVariant" PR.

When trying to run the tests from the root of godip I get an error about the gae module:
```gae/router/router.go:11:2: cannot find package "common"```
This means I can't run all the tests for the project with `go test ./...`.  Is there some other command you use to run all the tests?

**Comment by zond** [comment]
No, I just run them one at a time..

On May 23, 2017 08:03, "tttppp" <notifications@github.com> wrote:

Ok - I just ran quite a few of the tests and they all seem ok. This needs
to be merged at the same time as the diplicity "PureVariant" PR.

When trying to run the tests from the root of godip I get an error about
the gae module:
gae/router/router.go:11:2: cannot find package "common"
This means I can't run all the tests for the project with go test ./.... Is
there some other command you use to run all the tests?

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/zond/godip/pull/8#issuecomment-303298880>, or mute the
thread

**Comment by zond** [comment]
Does this mean you're ready to merge this PR?

**Comment by tttppp** [comment]
Yes, this is ready now. Thanks.

**Pull request merged**